xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/transformation/XWikiIconProvider.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.internal.transformation;
```

Comment:
Why is this in a `transformation` package? This is not really related to transformations anymore I think.

Reply:
The `DefaultIconProvider` from `xwiki-rendering` is in a transformation package. I want to inherit this class which is internal, so I figured I'd put it in the same package.
Moreover, there's no real place where this would suit perfectly, the best I could find from a quick look would be `org.xwiki.rendering.internal.util`.

For reference, the classes `XWikiHTMLRawBlockFilter` and `XWikiErrorBlockGenerator` are in a similar situation.

I'm not very confident in this choice, let me know where you'd put it instead.

___
Taking a look to see if it would make sense to fit the `DefaultIconProvider` in utils, and move this one afterwards. :eyes:

Reply:
From what I can see, there's no dependency anymore to anything in the transformation package, so I'll be moving those classes to `util` and updating the `components.txt` files :)

Reply:
Move done in https://github.com/xwiki/xwiki-platform/pull/2590/commits/e1dfba516e54c9fc40d6d909e2781097761b3a33 :+1: